src/pages/home/report/withReportAndReportActionOrNotFound.js

```diff
@@ -0,0 +1,147 @@
+        reportActions: PropTypes.shape(reportActionPropTypes),
+
+        /** The policies which the user has access to */
+        policies: PropTypes.objectOf(
+            PropTypes.shape({
+                /** The policy name */
+                name: PropTypes.string,
+
+                /** The type of the policy */
+                type: PropTypes.string,
+            }),
+        ),
+
+        /** Route params */
+        route: PropTypes.shape({
+            params: PropTypes.shape({
+                /** Report ID passed via route */
+                reportID: PropTypes.string,
+
+                /** ReportActionID passed via route */
+                reportActionID: PropTypes.string,
+            }),
+        }).isRequired,
+
+        /** Beta features list */
+        betas: PropTypes.arrayOf(PropTypes.string),
+
+        /** Indicated whether the report data is loading */
+        isLoadingReportData: PropTypes.bool,
+
+        /** Is the window width narrow, like on a mobile device? */
+        isSmallScreenWidth: PropTypes.bool.isRequired,
+    };
+
+    const defaultProps = {
+        forwardedRef: () => {},
+        reportActions: {},
+        report: {
+            isLoadingReportActions: true,
+        },
+        policies: {},
+        betas: [],
+        isLoadingReportData: true,
+    };
+
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportAndReportActionOrNotFound(props) {
+        // For small screen, we don't call openReport API when we go to a sub report page by deeplinnk
+        // So we need to call openReport here for small screen
+        const firstRef = useRef(true);
+        useEffect(() => {
+            if (!firstRef.current) {
+                return;
+            }
+            firstRef.current = false;
+            if (props.isSmallScreenWidth) {
+                Report.openReport(props.route.params.reportID);
+            }
+        }, [props.isSmallScreenWidth, props.route.params.reportID]);
+
+        const isLoadingReport = props.isLoadingReportData && (_.isEmpty(props.report) || !props.report.reportID);
+        const shouldHideReport = !isLoadingReport && (_.isEmpty(props.report) || !props.report.reportID || !ReportUtils.canAccessReport(props.report, props.policies, props.betas));
+
+        let reportAction = props.reportActions[`${props.route.params.reportActionID.toString()}`];
+        // Handle threads if needed
+        if (reportAction === undefined || reportAction.reportActionID === undefined) {
+            reportAction = ReportActionsUtils.getParentReportAction(props.report);
+        }
+        const isLoadingReportAction = _.isEmpty(props.reportActions) || (props.report.isLoadingReportActions && _.isEmpty(reportAction));
+
+        if ((isLoadingReport || isLoadingReportAction) && !shouldHideReport) {
+            return <FullscreenLoadingIndicator />;
+        }
+        if (shouldHideReport || _.isEmpty(reportAction)) {
```

Comment:
Please check my comment above and then I feel `shouldHideReport` won't be needed, the conditions can be put here inline.